src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2677,8 +2677,8 @@ TargetShardIntervalForFastPathQuery(Query *query, bool *isMultiShardQuery,
 	}
 
 	/* we're only expecting single shard from a single table */
-	Node *distKey PG_USED_FOR_ASSERTS_ONLY = NULL;
-	Assert(FastPathRouterQuery(query, &distKey) || !EnableFastPathRouterPlanner);
+	/*Node *distKey PG_USED_FOR_ASSERTS_ONLY = NULL; */
+	/*Assert(FastPathRouterQuery(query, &distKey) || !EnableFastPathRouterPlanner); */
```

Comment:
I would expect this to still hold / we shouldn't go in here for non-fast-path queries.

Reply:
this no longer holds because we rewrote the RTEs already, but we can remove the assert